src/EventTypeDefs.ts

```diff
@@ -106,28 +106,30 @@ export type TModificationEvents =
 
 export interface ModifiedEvent<E extends Event = TPointerEvent> {
   e?: E;
-  transform: Transform;
+  transform?: Transform;
   target: FabricObject;
   action?: string;
 }
 
-type ModificationEventsSpec<
-  Prefix extends string = '',
-  Modification = BasicTransformEvent,
-  Modified = ModifiedEvent | never
-> = Record<`${Prefix}${TModificationEvents}`, Modification> &
-  Record<`${Prefix}modified`, Modified>;
-
-type ObjectModificationEvents = ModificationEventsSpec;
-
-type CanvasModificationEvents = ModificationEventsSpec<
-  'object:',
-  BasicTransformEvent & { target: FabricObject },
-  // TODO: this typing makes not possible to use properties from modified event
-  // in object:modified
-  ModifiedEvent | { target: FabricObject }
-> & {
+type ObjectModificationEvents = {
+  moving: BasicTransformEvent;
+  scaling: BasicTransformEvent;
+  rotating: BasicTransformEvent;
+  skewing: BasicTransformEvent;
+  resizing: BasicTransformEvent;
+  modifyPoly: BasicTransformEvent;
+  modified: ModifiedEvent;
+};
+
+type CanvasModificationEvents = {
   'before:transform': TEvent & { transform: Transform };
+  'object:moving': BasicTransformEvent & { target: FabricObject };
+  'object:scaling': BasicTransformEvent & { target: FabricObject };
+  'object:rotating': BasicTransformEvent & { target: FabricObject };
+  'object:skewing': BasicTransformEvent & { target: FabricObject };
+  'object:resizing': BasicTransformEvent & { target: FabricObject };
+  'object:modifyPoly': BasicTransformEvent & { target: FabricObject };
+  'object:modified': ModifiedEvent;
```

Comment:
for me those types are more readable than the previous one. helps me keeping them in check.

Reply:
Definitely better